libs/web/components/src/MarkdownField.tsx

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import { FC, useCallback, useState } from "react";
+import { classNames } from "react-extras";
+
+import { IntrinsicElements } from "@app-gov/web/types";
+
+import { Markdown } from "./";
+
+interface MarkdownFieldProps {}
+
+export const MarkdownField: FC<
+	IntrinsicElements["textarea"] & MarkdownFieldProps
+> = ({ className, children, value, rows = 10, ...props }) => {
+	const [preview, setPreview] = useState<boolean>(false);
+	const onWriteClick = useCallback(() => {
+		setPreview(false);
+	}, []);
+
+	const onPreviewClick = useCallback(() => {
+		setPreview(true);
+	}, []);
+
```

Comment:
why use callbacks over inline here?

Reply:
just OCD really, inline creates new function every time, meaning the component receiving that prop will  be also re-render even though it technically doesn't change anything